bpf/ingress_node_firewall_kernel.c

```diff
@@ -417,20 +436,20 @@ ingress_node_firewall_main(struct xdp_md *ctx) {
 
     switch (action) {
     case DENY:
-        generate_event_and_update_statistics(ctx, (__u16)(dataEnd - data), DENY, ruleId, 1, ifId);
-        bpf_printk("Ingress node firewall action DENY -> XDP_DROP");
+        generate_event_and_update_statistics(ctx, bpf_xdp_get_buff_len(ctx), DENY, ruleId, 1, ifId);
+        ingress_node_firewall_printk("Ingress node firewall action DENY -> XDP_DROP");
         return XDP_DROP;
     case ALLOW:
-        generate_event_and_update_statistics(ctx, (__u16)(dataEnd - data), ALLOW, ruleId, 0, ifId);
-        bpf_printk("Ingress node firewall action ALLOW -> XDP_PASS");
+        generate_event_and_update_statistics(ctx, bpf_xdp_get_buff_len(ctx), ALLOW, ruleId, 0, ifId);
```

Comment:
The use of this helper will make the program fail to load on older kernels that don't have the helper available. Just want to double-check that this is OK?

Reply:
Yes well aware of this

Reply:
Alright, cool. In that case, LGTM :)